aptos-move/block-executor/src/executor.rs

```diff
@@ -322,9 +294,6 @@ where
                         !matches!(scheduler_task, SchedulerTask::ExecutionTask(_, _)),
```

Comment:
This is now the only reason we pass scheduler_task. I think the method can return something if it commits the last thing, and we can assert at the caller (we could get rid of it too, but the assert seems somewhat useful)